app/views/layouts/application.html.erb

```diff
@@ -4,7 +4,8 @@
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
 
-    <%= controller.ssr_head %>
+    <%= ssr_head %>
+    <%= ssr_data_tag %>
```

Comment:
what is that for?

Reply:
Those are the props and the view name which are necessary to hydrate the view. react-rails places this as data-attributes inside the root div:

```
<div data-react-class="some/component" data-react-props="{&quot;csrfToken&quot;:&quot;af77RKYOafgH6sSFjvEQMeaMz3u...>
```

This can get pretty large, which pollutes the element inspector (bad developer experience):

![image](https://user-images.githubusercontent.com/4227520/67765290-0d5a7000-fa4c-11e9-8907-0bb54fbdccd3.png)

(pixelated to remove sensitive informations)

Now this data is placed inside a `<script type="application/json">` tag, and as a normal unescaped json string (also better developer experience, and less document size).

It's also similar to other react ssr frameworks like Next.js